pulpcore/cli/common/generic.py

```diff
@@ -370,6 +422,39 @@ def _multi_option_callback(
     return click.option(*args, **kwargs)
 
 
+def type_option(*args: Any, **kwargs: Any) -> Callable[[_FC], _FC]:
+
+    choices: Dict[str, Type[PulpEntityContext]] = kwargs.pop("choices")
+    assert choices and isinstance(choices, dict)
+    type_names = list(choices.keys())
+    case_sensitive = kwargs.pop("case_sensitive", False)
+    defaults = {
+        "cls": PulpOption,
+        "default": type_names[0],
+        "is_eager": True,
+        "expose_value": False,
+    }
+
+    def _type_callback(ctx: click.Context, param: click.Parameter, value: Optional[str]) -> str:
+        pulp_ctx = ctx.find_object(PulpContext)
+        assert pulp_ctx
+        if value in choices:
```

Comment:
I think we can assume click will provide either one choice or `None`.
```suggestion
        if value is not None:
```
And `None` should not even be possible given that we have a default.

Reply:
Technically you can still override the default to be a value not inside choices, but I haven't tested this to see if it errors or not.